### DISCUSSION TITLE
Use partials in the pages folder

### DISCUSSION THREAD

**Discussion by dile1l1** [joomlatools/joomlatools-pages#894] [Q&A] [answered]
<img width="745" alt="CleanShot 2022-05-13 at 16 00 39@2x" src="https://user-images.githubusercontent.com/38931020/168238730-be30e156-fb1f-4afb-bc77-2813c7a2a157.png">
<img width="424" alt="CleanShot 2022-05-13 at 16 00 54@2x" src="https://user-images.githubusercontent.com/38931020/168239531-3971fbd6-f640-4ba0-954f-acae110c79d1.png">
I know partials normally import from the templates/partials folder due to the wiki. <br>
but is it possible to import form the same folder as pages located, which show as the screenshot.

intro.html is the partials I want to import into index.html.php

**Comment by johanjanssens** [comment, marked as answer]
Yes, use:

```php
<?= import('intro.html') ?>
```

For more info see: https://github.com/joomlatools/joomlatools-pages/pull/2  

Note that you can also load anything from your `/theme` folder using `import('theme:/path/to/file)`see: https://github.com/joomlatools/joomlatools-pages/pull/12

**Reply by dile1l1** [reply]
Get Internal Server Error error page after apply your answer
We have set configuration-pages.php to change our root folder name from joomlatools-pages to sites
```php
<?php

return [
    'sites' => [
        '[*]' => JPATH_ROOT.'/sites',
    ],
];
```
Could this be a possible reason for the error?

**Reply by johanjanssens** [reply]
Can you please turn on debug and set error reporting to max. This should give you a more meaningfull error. Make a screenshot of that and post it. Also please post a screenshot of your folder structure, if you changes your default conf it's good to know exactly what you changed it too so can help you. The more info you can provide the easier and quicker to assist you.

**Reply by dile1l1** [reply]
## folder name change
As said before, we replaced the original folder name joomlatool-pages with sites
![CleanShot 2022-05-18 at 12 50 38@2x-edit](https://user-images.githubusercontent.com/38931020/168959954-2262357b-687e-4ced-9e00-b5692838a231.png)


## pages folder structure
```
sites/pages/tw/test/
├── index.html.php
└── intro.html
```
### pages content
```
<!-- in index.html.php -->
<?= import('intro.html') ?>
<div class="test">haha</div>

<!-- in intro.html -->
<h3>test</h3>
<div>
   test123
</div>
```
after apply `<?= import('intro.html') ?>` , page would get Internal Server Error 500
here is the screenshot，and the debug info
![CleanShot 2022-05-18 at 13 00 52@2x](https://user-images.githubusercontent.com/38931020/168961240-6934eafa-4c6d-4b42-852e-c1e9e8239c7d.png)
![dev gsscloud-0426_tw_test](https://user-images.githubusercontent.com/38931020/168961277-135c1374-3756-41f2-b0a8-a0555fdf6a48.png)

## config.php
```php
<?php

return [

    'http_cache'               => false,
    'http_cache_path'          => JPATH_ROOT.'/cache/joomlatools-pages',
    'http_cache_time'          => false,
    'http_cache_time_browser'  => null,
    'http_cache_validation'      => true,
    'http_cache_control'         => [],
    'http_cache_control_private' => ['private', 'no-cache'],

    //Static caching
    'http_static_cache'         => getenv('PAGES_STATIC_ROOT') ? true : false,
    'http_static_cache_path'    => getenv('PAGES_STATIC_ROOT') ? getenv('PAGES_STATIC_ROOT') : false,

    //Page caching
    'page_cache'            => true,
    'page_cache_path'       => JPATH_ROOT.'/cache/joomlatools-pages/pages',
    'data_cache_validation' => true,

    //Data caching
    'data_cache'            => false,
    'data_cache_path'       => JPATH_ROOT.'/cache/joomlatools-pages/data',
    'data_cache_validation' => true,
    'data_namespaces'       => [],

    //Remote caching
    'http_resource_cache'        => JFactory::getConfig()->get('caching'),
    'http_resource_cache_path'   => JPATH_ROOT.'/cache/joomlatools-pages/resources',
    'http_resource_cache_debug'  => JDEBUG ? false : true,

    //Template caching
    'template_cache'        => JDEBUG ? false : true,
    'template_cache_path'   => JPATH_ROOT.'/cache/joomlatools-pages/templates',
    'template_cache_validation' => true,

    // Site
    'site' => [
        'name'  => 'Vital雲端服務 - 叡揚資訊',
    ],

    // Page
    'page' => [

        'metadata' => [
            'generator'     => 'mock',
            'author'        => 'mock',
            'robots'        => 'index, follow',
            'summary'       => '',
            'og:site_name'  => '',
            'og:url'        => '',
            'og:title'      => '',
            'og:description'=> '',
            'og:image'      => 'theme://images/og/home.jpg',
            'twitter:site'  => '',
            'twitter:card'  => '',
        ],

        'visible'   => true,
        'published' => true,
    ],

    // Aliases
    'aliases' => [
        'theme://'  => '/sites/theme/',
        'pages://'  => '/sites/pages/',
        'templates://'  => '/templates/vital2022/',
    ],

    // Google Analytics
    'ga_code' => '',

    // Extensions
    'extension_config' => [

        'ext:joomla.model.articles'  => [
            'aliases' => [
                'about-us' => 1,
            ],
        ],
    ],
];

```

**Reply by dile1l1** [reply]
@johanjanssens Is the above information sufficient to determine the problem?

**Reply by johanjanssens** [reply]
I didn't have the time yet to look into this. Been very hard at work to finish https://github.com/joomlatools/joomlatools-server to make it easier to offer community support.

If you check the error message generated by our build in debugger you will see that this is not an error with Pages but a specific captcha plugin you are using. Its using deprecated code, depening on your exact version of PHP deprecating notices in PHP will be warnings or errors, which would explain the problem.

Question. What is the version of PHP you are using? 

### Next steps

Next, to make it easier to assist you, the best approach would be:

1. Use Joomlatools Server for local dev, and first make sure it works in there, if it does we would know it's a local setup issue on your end.

2. Put all your code in Github, if you put the whole Joomla site in Github, we can run it in [Gitpod](https://gitpod.io) easily and check for issues. As you can see some other extensions can also interfere so we would need a complete working site with everything you are running. (*)
